src/govuk/components/header/template.njk

```diff
@@ -54,9 +54,10 @@
     </a>
     {% endif %}
     {% if params.navigation %}
-    <button type="button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="{{ params.menuButtonLabel | default('Show or hide navigation menu') }}">Menu</button>
-    <nav>
-      <ul id="navigation" class="govuk-header__navigation {{ params.navigationClasses if params.navigationClasses }}" aria-label="{{ params.navigationLabel | default('Navigation menu') }}">
+    <nav aria-label="{{ params.navigationLabel | default('Menu') }}">
+      <button type="button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="{{ params.menuButtonLabel | default('Show or hide menu') }}">Menu</button>
+
+      <ul id="navigation" class="govuk-header__navigation {{ params.navigationClasses if params.navigationClasses }}" >
```

Comment:
I've left the `__navigation` element and any passed `navigationClasses` on the `<ul>`. There's an argument for moving them to the `<nav>` as well, but I tried to make the minimum set of changes to fix the issue. Happy to revisit though – it does seem odd to have the `__navigation` element not on the `<nav>`.

Reply:
Hm, tricky 🤔 Like you said, it seems a bit odd to be applying them to the `<ul>`. But I'm also not sure what people are using the `navigationClasses` for, which makes it difficult to know if they'd expect/find it useful for them to be applied to the `<ul>` or the `<nav>`. If we move it to the `<nav>` element, will we also need to provide a way for people to pass classes to the `<ul>`?

I'm not sure if we can separate the two - move the `govuk-header__navigation` class to the `<nav>` but keep the `navigationClasses` on the `<ul>`? Perhaps that's even more confusing though.

Reply:
Yeah, I think it'd make sense to keep the two together.

Arguments for keeping `govuk-header__navigation` and `navigationClasses` on the `<ul>`:

- Preserves existing behaviour
- We don't know what people are using `navigationClasses` for, so it's hard to reason about what impact the change would have

Arguments for moving `govuk-header__navigation` and `navigationClasses` to the `<nav>`:

- Seems more 'correct' / what users would expect
- Users can still style the list if they need to by passing e.g. `app-header__navigation` as a `navigationClass` and then using `.app-header__navigation ul` to target it, but without this change would have no equivalent way to style the nav
- We could make it possible to add classes to the `<ul>` by introducing a new param, without having to make a breaking change
- If we didn't do this and later wanted to make it possible to style the `<nav>` we'd have to introduce a new param with a suboptimal name (e.g. `navigationWrapperClasses`) or come back and make this (breaking) change

Reply:
Thanks for summarising, @36degrees, that's really helpful! I don't think you've missed anything. It's a really good point about being able to style the `<ul>` by using the `navigationClasses` but not being able to do it the other way around - I think that's convinced me that we should make the change. I think we might need to provide some additional guidance or examples somewhere to help people make the change though, maybe in the Changelog

Reply:
Yeah, I think I've talked myself into it too. Will push a commit…